### PR TITLE
tests: bump CockroachDB container image to version 2.1.7

### PR DESCRIPTION
**Pull request by jgehrcke**
This is to see if this fails non-trivially, or if it maybe even passes!

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 alembic==1.0.5
 Cython==0.29.1
-cockroachdb==0.2.1
+cockroachdb==0.3.1
```
Comment (gpaul) [inline requirements.txt]: I'm most curious about what changed in this client library - might be worth looking at the commit log.

Comment (jgehrcke) [reply requirements.txt]: I did, actually. One of the commit messages in this PR is:

```
Addresses cockroachdb/cockroachdb-python#73

Specifically, w/o this we see

  dialect.py:138: SAWarning: Could not parse type name 'character varying'

and subsequently

  TypeError: 'NullType' object is not callable

while

  generate_migration.py] INFO: Load initial schema into DB and apply known migrations

This appears during

  make check-generated-files-up-to-date
```

Comment (jgehrcke) [reply requirements.txt]: (not a lot changed, and the new version fixes cockroachdb/cockroachdb-python#73 which is what we seem to need)

Comment (jgehrcke) [reply requirements.txt]: Changes: https://github.com/cockroachdb/cockroachdb-python/compare/9d66a14322e6365df6189e88394ea555d22f137c...590f5ee8f488da87e246bf85efae0e3c83f4766a

Changelog:

```
Version 0.3.1
Released Feb 25, 2019

Support CockroachDB version numbers greater than 2.
Version 0.3.0
Released Jan 23, 2019

Added support for more data types.
Improved introspection of types with modifiers (decimal, varchar).
Improved introspection of unique constraints.
```

**Pull request merged**